sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -224,6 +224,8 @@ class Dataset[T] private[sql](
     // For various commands (like DDL) and queries with side effects, we force query execution
     // to happen right away to let these side effects take place eagerly.
     val plan = queryExecution.analyzed match {
+      case _: V2MicroBatchWriteCommand =>
```

Comment:
This bit is needed to prevent executing each batch early. We used `WriteToDataSourceV2` before, which did not extend `Command`.